osu.Game/Beatmaps/BeatmapManager.cs

```diff
@@ -94,6 +95,34 @@ public partial class BeatmapManager : DownloadableArchiveModelManager<BeatmapSet
 
         protected override bool ShouldDeleteArchive(string path) => Path.GetExtension(path)?.ToLowerInvariant() == ".osz";
 
+        public WorkingBeatmap CreateNew(RulesetInfo ruleset)
+        {
+            var metadata = new BeatmapMetadata
+            {
+                Artist = "artist",
+                Title = "title",
+                Author = User.SYSTEM_USER,
+            };
+
+            var set = new BeatmapSetInfo
+            {
+                Metadata = metadata,
+                Beatmaps = new List<BeatmapInfo>
+                {
+                    new BeatmapInfo
+                    {
+                        BaseDifficulty = new BeatmapDifficulty(),
+                        Ruleset = ruleset,
+                        Metadata = metadata,
+                        Version = "difficulty"
+                    }
+                }
+            };
+
+            var working = Import(set).Result;
```

Comment:
Because of this a new beatmap is imported and stored every time the editor is entered and exited, even if nothing of value was actually done. Worse yet, because the author is set to the current user in `Editor`, the beatmap looks fine in the run after it was created as it stays in memory, but after a restart the change is lost and the author rolls back to `system`.

Reply:
I think moving the user providing to the ctor is a good solution for now. This will likely change going forward (to the point of maybe not even importing to the database until first editor save? not sure, we'll see the direction things take).